src/Fhir.Anonymizer.Core/Extensions/ElementNodeOperationExtensions.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Fhir.Anonymizer.Core.AnonymizationConfigurations;
+using Fhir.Anonymizer.Core.Models;
+using Fhir.Anonymizer.Core.Processors;
+using Fhir.Anonymizer.Core.Visitors;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Specification;
+
+namespace Fhir.Anonymizer.Core.Extensions
+{
+    public static class ElementNodeOperationExtensions
+    {
+        private static readonly PocoStructureDefinitionSummaryProvider s_provider = new PocoStructureDefinitionSummaryProvider();
+
+        public static ElementNode Anonymize(this ElementNode node, AnonymizationFhirPathRule[] rules, Dictionary<string, IAnonymizerProcessor> processors)
+        {
+            AnonymizationVisitor visitor = new AnonymizationVisitor(rules, processors);
+            node.Accept(visitor);
+            node.RemoveNullChildren();
+
+            return node;
+        }
+
+        public static void RemoveNullChildren(this ElementNode node)
+        {
+            if (node == null)
+            {
+                return;
+            }
+
+            var children = node.Children().Cast<ElementNode>().ToList();
+            foreach (var child in children)
+            {
+                RemoveNullChildren(child);
+            }
+
+            if (!node.Children().Any() && node.Value == null && !Enum.TryParse<ResourceType>(node.InstanceType, true, out _))
+            {
```

Comment:
What is this check for? Any comment with example to make it clear?

Reply:
Refact this method. We would only remove one node if currentNodeIsEmpty & !currentNodeIsFhirResource